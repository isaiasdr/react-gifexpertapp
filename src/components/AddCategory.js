import React, { useState } from 'react';
import propTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories( categories => { 
                if(!categories.find( categorie => categorie === inputValue))    
                    return [ inputValue, ...categories ];

                else
                    return [...categories];
            });

            setInputValue( '' );
        }
    }

    return (
        <form onSubmit = { handleSubmit }>
            <input 
                type="text"
                value = { inputValue }
                onChange = { handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: propTypes.func.isRequired
};