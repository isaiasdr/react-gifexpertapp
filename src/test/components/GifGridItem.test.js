import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test in Component GifGridItem', () => {

    const title = "Prueba";
    const url = "https://localhost.test.jpg";

    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('should show GifGridItem correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a paragraph with the title', () => {
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
    });

    test('should have a image equal to url from props', () => {
        const img = wrapper.find('img');

        expect( img.props().src ).toBe( url );
    });
    
    test('should have a image with alt to title', () => {
        const img = wrapper.find('img');

        expect( img.props().alt ).toBe( title );
    });

    test('should have a div with class animate__fadeIn', () => {
        const div = wrapper.find('div');
        
        expect( div.props().className ).toMatch( 'animate__fadeIn' );
    });
});