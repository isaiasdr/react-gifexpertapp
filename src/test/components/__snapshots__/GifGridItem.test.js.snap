// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Component GifGridItem should show GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Prueba"
    src="https://localhost.test.jpg"
  />
  <p>
    Prueba
  </p>
</div>
`;
